### PR TITLE
Support reanalyze examples

### LINKED ISSUE #68: Add reanalyze repository to `test_wild`
Hey @nkrkv,

during #67 I try to add reanalyze repository to the test, and it had several errors in it
So it could be interesting to add it to the parser

Thanks for maintained this treesitter grammar :) 

Cheers

(I don't have time to take a look at it for now, but will when I will have more time)

### LINKED ISSUE #171: Error: consecutive `and` let bindings
Still getting some errors even after #136 

Reanalyze examples: See [TestCyberTruck.res](https://github.com/rescript-association/reanalyze/blob/master/examples/termination/src/TestCyberTruck.res)

```
yarn tree-sitter parse --quiet --stat ../tree-sitter-test-rescript/reanalyze/examples/termination/src/TestCyberTruck.res
```

```
../tree-sitter-test-rescript/reanalyze/examples/termination/src/TestCyberTruck.res	11 ms	(ERROR [197, 0] - [366, 1])
Total parses: 1; successful parses: 0; failed parses: 1; success percentage: 0.00%
```

### PR DESCRIPTION
**Pull request by aspeddro**
Close #171 
Close #68

### CODE REVIEW

File: grammar.js [hunk 1]
```diff
@@ -468,7 +468,7 @@ module.exports = grammar({
         '=',
         repeat($.decorator),
         $.expression,
-        repeat(alias($._let_binding_and, $.let_binding)),
+        optional(alias($._let_binding_and, $.let_binding)),
```
Comment (aspeddro) [inline grammar.js:471]: I have no clear idea why it is generating an error (#171).

My hypothesis: As the rule `_let_binding_and` has a `_let_binding` rule so it's already _recursive_.

Comment (nkrkv) [reply grammar.js:471]: Perhaps, yes. I also see no precedence specs for `let_binding`, `_let_binding`, `_let_binding_and`. Maybe that’s what involved too.

Nevertheless, I think the new variant is fine too.

File: test/corpus/let_bindings.txt [hunk 2]
```diff
@@ -257,11 +257,11 @@ and b = c
         (decorator (decorator_identifier))
         (decorator (decorator_identifier))
         (value_identifier)
-        (value_identifier))
+        (value_identifier)
       (let_binding
         (decorator (decorator_identifier))
         (value_identifier)
-        (number)))))
+        (number))))))
```
Comment (nkrkv) [inline test/corpus/let_bindings.txt:264]: The third `let_binding` is not a child of the second `let_binding`. So, would you indent this block one step right to make this apparent?

**Review by nkrkv — CHANGES_REQUESTED** [review]

**Review by nkrkv — APPROVED** [review]
:partying_face:

**Pull request merged**